scripts/Packages/Package_BGA/bga.py

```diff
@@ -153,6 +153,19 @@ def crtYdRound(x):
 
     # Pads
     balls = layoutX * layoutY
+    
+    # use IPC-derived pad size if possible, then fall back to user-defined pads
+    if (("ball_type" in fpParams) and ("ball_diameter" in fpParams)):
+        padSize = configuration['collapsible'][fpParams["ball_diameter"]]["min"]
+        padSize = [padSize, padSize]
+        if "pad_diameter" in fpParams:
+            print("Pad size is being derived using IPC rules even though pad diameter is defined.")
+    elif "pad_diameter" in fpParams:
+        padSize = [fpParams["pad_diameter"], fpParams["pad_diameter"]]
+    else:
+        print("\033[1;35mThe config file must include 'ball_type' and 'ball_diameter' or 'pad_diameter'. Exiting...\033[0m")
```

Comment:
I don't really like the escape-codes. It is hard to read. At least I don't know out of the back of my head which colors that are.
A similar print() is also used in `scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/ipc_noLead_generator.py` where the same comment applies.

Perhaps add a convenience function like `print_color()`

Reply:
True. I switched to a `KeyError`, which applies well. Adding that helper function for printing in color would be nice. I'll do that later.

Reply:
Tangent:
Python probably has some kind of termcap/terminfo support.
How about only outputting color when supported by the output device?